cairo-sys-rs/src/lib.rs

```diff
@@ -742,6 +747,22 @@ extern "C" {
     pub fn cairo_text_cluster_allocate(num_clusters: c_int) -> *mut TextCluster;
     pub fn cairo_text_cluster_free(clusters: *mut TextCluster);
 
+    #[cfg(any(feature = "freetype", feature = "dox"))]
+    pub fn cairo_ft_font_face_create_for_ft_face(
+        face: FT_Face,
+        load_flags: c_int,
+    ) -> *mut cairo_font_face_t;
+    #[cfg(any(feature = "freetype", feature = "dox"))]
+    pub fn cairo_ft_font_face_create_for_pattern(pattern: *mut FcPattern) -> *mut cairo_font_face_t;
+    #[cfg(any(feature = "freetype", feature = "dox"))]
+    pub fn cairo_ft_font_option_substitute(
+        options: *const cairo_font_options_t,
+        pattern: *mut FcPattern,
+    );
+    #[cfg(any(feature = "freetype", feature = "dox"))]
+    pub fn cairo_ft_scaled_font_lock_face(scaled_font: *mut cairo_scaled_font_t) -> FT_Face;
+    #[cfg(any(feature = "freetype", feature = "dox"))]
+    pub fn cairo_ft_scaled_font_unlock_face(scaled_font: *mut cairo_scaled_font_t);
```

Comment:
The two functions above could get save bindings though, only the others with the `*mut c_void` don't make much sense

Reply:
I'm sorry, what do you mean by "save bindings"?

Reply:
`s/save/safe/`, sorry. The non-`sys` ones.

Reply:
Oh sorry, you mean "safe bindings". I will think about what this would look like.

Reply:
Looks like something like a `MutexGuard`-style guard type can be used here

Reply:
There might be a problem with defining safe methods here: the functions expect a font face object created from freetype, but this isn't enforced in the type system. If a non-freetype font is passed to these functions I don't know what will happen.

Reply:
We could track whether the `FontFace` in rust was created from freetype, possibly using zero-sized types to make it zero cost, but it would add complexity.

Reply:
Can that be checked at runtime and be handled via a `panic!`? Otherwise mark the functions `unsafe`

Reply:
Yes there is the possibility of enforcing the locking rules at compile time using Rust's ownership model. I would need to think through all the details of this as the ownership models of freetype and cairo are different and interact in potentially complex ways. It might be safer to let the user handle this unless we are sure we can provide a safe abstraction.

Reply:
Let's worry about that some other time then

Reply:
I tried adding in a marker value for tracking font type at runtime, but it doesn't work with `glib_wrapper!`, so some work would be required in `glib-rs` to make it possible. I think it's probably best to leave it for now as you say.